lib/OpenQA/Schema/Schema.pm

```diff
@@ -24,12 +25,22 @@ use FindBin qw($Bin);
 # after bumping the version please look at the instructions in the docs/Contributing.asciidoc file
 # on what scripts should be run and how
 our $VERSION = 24;
+our $configname="$Bin/../lib/database.ini";
 
 __PACKAGE__->load_namespaces;
 
-my @paths = ( "$Bin/../lib/database", "$Bin/../../lib/database" );
-unshift(@paths, dirname($ENV{OPENQA_CONFIG}).'/database') if ($ENV{OPENQA_CONFIG});
-__PACKAGE__->config_paths(\@paths);
+
+sub connect_db {
+    my $mode = shift || $ENV{OPENQA_DATABASE} || 'production';
+    CORE::state $schema;
+    unless ($schema) {
+        my %ini;
+        #        print("ENV{OPENQA_DATABASE_CONFIG}=$ENV{OPENQA_DATABASE_CONFIG}\n");
+        tie %ini, 'Config::IniFiles', ( -file => $ENV{OPENQA_DATABASE_CONFIG} || $configname );
+        $schema=__PACKAGE__->connect($ini{$mode});
```

Comment:
>
> Yeah, but is it needed? I don't think they are used anywhere in tests.
>
> yes, it is needed, for example t/api/06-products.t needs database.ini
>
> Ah right, by the very same thing you are trying to get rid of I suppose
(Schema::Config). I wasn't paying attention and it is hidden deep. :)
I expected tests to be self contained and if needed, database.ini would be
already there.